pvlib/iotools/tmy.py

```diff
@@ -150,24 +150,30 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
 
     References
     ----------
-
     .. [1] Wilcox, S and Marion, W. "Users Manual for TMY3 Data Sets".
        NREL/TP-581-43156, Revised May 2008.
-
     .. [2] Wilcox, S. (2007). National Solar Radiation Database 1991 2005
        Update: Users Manual. 472 pp.; NREL Report No. TP-581-41364.
-    '''
-
+    .. [3] `SolarAnywhere file formats
+       <https://www.solaranywhere.com/support/historical-data/file-formats/>`_
+    """  # noqa: E501
     head = ['USAF', 'Name', 'State', 'TZ', 'latitude', 'longitude', 'altitude']
-
-    with open(str(filename), 'r') as csvdata:
-        # read in file metadata, advance buffer to second line
-        firstline = csvdata.readline()
+    try:
+        fbuf = open(str(filename), 'r')
+        firstline = fbuf.readline()  # first line contains the metadata
+    # SolarAnywhere files contain non-UTF8 characters and require
+    # encoding='iso-8859-1' on Linux in order to be parsed
```

Comment:
I don't think this is particular to linux. I haven't tested this particular PR or file, but from other work I know that I need to specify the encoding when using solar anywhere files on my windows machine.

Reply:
The Linux part has been removed.